torchqdynamics/solver_options.py

```diff
@@ -1,14 +1,15 @@
 from dataclasses import dataclass
+from typing import Optional
 
 
 @dataclass
 class SolverOption:
-    pass
+    verbose: bool = True
 
 
 @dataclass
 class FixedStep(SolverOption):
-    dt: float
+    dt: Optional[float] = None
```

Comment:
Why is this optional? Shouldn't it always be passed by the user?

Reply:
Because of the way dataclass inheritance is made, now that `SolverOption` has a keyword argument (`verbose`) we **have to** use kw arguments only in the child classes.

> Note that all solver options must be keyword arguments because the base class has a kw args (csqce of dataclasses > https://medium.com/@aniscampos/python-dataclass-inheritance-finally-686eaf60fbb5).

See also commit description 228276c191209315f477897354787728464117ff.

Reply:
Because of the way dataclass inheritance is made, now that `SolverOption` has a keyword argument (`verbose`) we **have to** use kw arguments only in the child classes.

> Note that all solver options must be keyword arguments because the base class has a kw args (csqce of dataclasses > https://medium.com/@aniscampos/python-dataclass-inheritance-finally-686eaf60fbb5).

See also commit description 228276c191209315f477897354787728464117ff.

Reply:
In this case we need to add a check in the `__init__` of fixed time step solvers for None-like `dt`values, and raise an error in case `dt` is not defined.

Reply:
Actually I think we should not use dataclasses, see new commit.

Reply:
(it's very confusing for the user to see a field being default initialized to `None`, but then the code yells at him because he didn't set its value 😅 )